server/src/main/java/com/genymobile/scrcpy/DesktopConnection.java

```diff
@@ -60,40 +105,76 @@ private static String getSocketName(int scid) {
         return SOCKET_NAME_PREFIX + String.format("_%08x", scid);
     }
 
-    public static DesktopConnection open(int scid, boolean tunnelForward, boolean video, boolean audio, boolean control, boolean sendDummyByte)
+    public static DesktopConnection open(int scid, boolean tunnelForward, boolean video, boolean audio, boolean control, boolean sendDummyByte, int serverPort)
             throws IOException {
         String socketName = getSocketName(scid);
 
         LocalSocket videoSocket = null;
         LocalSocket audioSocket = null;
         LocalSocket controlSocket = null;
+
+        Socket videoExtSocket = null;
+        Socket audioExtSocket = null;
+        Socket controlExtSocket = null;
```

Comment:
I believe this is not a good design, which adds unnecessary complexity and redundant branching judgments to the program. In order to implement Socket support, I'd be inclined to implement a Socket wrapper to provide a unified interface, and subsequently implement Socket and LocalSocket separately within it. however, your code contribution here is definitely a good enough PoC.